### DISCUSSION TITLE
tuiTextfield show error without defocus

### DISCUSSION THREAD

**Discussion by vytautas-pranskunas-** [taiga-family/taiga-ui#9598] [Q&A] [answered]
Hi,

i have async validator which validates input. Error it self appears immediatly because of tui-error control however input itelf is not marked red untill i not explicilty on submit calling tuiMarkControlAsTouchedAndValidate
Also when tuiMarkControlAsTouchedAndValidate is called field marked as red only when field itslef is defocused
1) is there a standard way of marking it red without calling tuiMarkControlAsTouchedAndValidate
2) is there a way to mark it red without defocusing

https://github.com/user-attachments/assets/3c114d2e-1ad3-462a-85fb-bfa27b7ad6d8

**Comment by waterplea** [comment, marked as answer]
You can mark all your controls as touched from the beginning immediately after you create them. Otherwise it is not configurable — Taiga UI only shows errors for touched controls.
